### ISSUE TITLE
Disabled toolbar button icon slightly misrendered under wxMSW 3.2.0

### ISSUE THREAD

**Issue by dhansel** [wxWidgets/wxWidgets#22754]
**Describe the bug**

The *disabled* icon of a bitmap button in a toolbar under wxMSW is noticeably misrendered.

**Expected vs observed behaviour**

When specifying the same bitmap for both enabled and disabled state of a button, the button's icon should
be identical in enabled or disabled state. However, in the *disabled* state, the button is icon slightly different
(slightly misrendered). See the screenshots.png file in the attached [icon_issue.zip](https://github.com/wxWidgets/wxWidgets/files/9446371/icon_issue.zip):
- Icon (a) is the button in *disabled* state when using wxMSW 3.2.0
- Icon (b) is the button in *enabled* state when using wxMSW 3.2.0
- Icon (c) is the button in *disabled* state when using wxMSW 3.1.4
- Icon (d) is the button in *enabled* state when using wxMSW 3.1.4

When using wxMSW 3.1.4, the behavior is as expected but it it incorrect in wxMSW 3.2.0

**Patch or snippet allowing to reproduce the problem**

The screen shots above were created by adding the following lines into the wxWidgets "minimal" 
sample (at the end of the MyFrame constructor):

```
#if 0
    wxBitmapBundle bmp = wxBitmapBundle::FromIconBundle(wxIconBundle("testicon", 0));
#else
    wxIcon icon;
    wxBitmap bmp;
    icon.LoadFile("testicon", wxBITMAP_TYPE_ICO_RESOURCE, 24, 24);
    bmp.CopyFromIcon(icon);
#endif

    wxToolBar* toolbar = new wxToolBar(this, -1, wxDefaultPosition, wxDefaultSize, wxNO_BORDER | wxTB_HORIZONTAL);
    toolbar->AddTool(1, _("test1"), bmp, bmp);
    toolbar->AddTool(2, _("test2"), bmp, bmp);
    toolbar->EnableTool(1, false);
    toolbar->EnableTool(2, true);
    toolbar->Realize();
    SetToolBar(toolbar);
```

In addition to the code, the .ico file and a modified .rc file are required to compile. 
I have attached a ZIP file containing all files necessary to compile under wxMSW.

[icon_issue.zip](https://github.com/wxWidgets/wxWidgets/files/9446371/icon_issue.zip)

Note that I also tried using the new wxBitmapBundle functions to specify the button
icon but the results are the same.

**To Reproduce**

Compile the attached modified version of the minimal sample. Start the program
and notice that the two toolbar buttons are rendered differently.

**Platform and version information:**

 - wxWidgets (wxMSW) 3.2.0 
 - Visual Studio 2022
 - Windows Version 10.0.19044 Build 19044

**Comment by dhansel** [comment]
I found the following code in function wxToolbar::Realize() within src/msw/toolbar.cpp:

```
#if wxUSE_WXDIB
        if ( bitmap.HasAlpha() )
        {
            // Strangely, toolbar expects bitmaps with transparency to not
            // be premultiplied, unlike most of the rest of win32. Without this
            // conversion, e.g. antialiased lines would be subtly, but
            // noticeably misrendered.
            hBitmap = wxDIB(bitmap.ConvertToImage(),
                            wxDIB::PixelFormat_NotPreMultiplied).Detach();
        }
        else
#endif
```

If I comment out this code then **both** enabled and disabled button icons
are misrendered. It appears that whatever this code does to the enabled icon
should also be done to the disabled icon.

**Comment by vadz** [comment]
I can indeed see this, thanks for reporting. I'll take a look.

**Comment by vadz** [comment]
As I suspected, this is yet another consequence of my `wxImageList` changes... Should be fixed by #22756, which I'll merge soon and backport to 3.2 too.

Thanks again for reporting!

**Comment by dhansel** [comment]
Thanks for the quick fix! I applied your changes to my local copy of wxWidgets 3.2.0 here and now 
all toolbar buttons look correct.
